lib/Sema/TypeCheckStorage.cpp

```diff
@@ -1494,6 +1494,7 @@ static std::pair<BraceStmt *, bool>
 synthesizeObservedSetterBody(AccessorDecl *Set, TargetImpl target,
                              ASTContext &Ctx) {
   auto VD = cast<VarDecl>(Set->getStorage());
+  bool isLazy = !VD->isStatic() && VD->getAttrs().hasAttribute<LazyAttr>();
```

Comment:
Can this be a parameter to synthesizeObservedSetterBody() instead, since the caller always knows if its dealing with a lazy property or not?